core/src/main/java/nl/nn/adapterframework/webcontrol/api/TransactionalStorage.java

```diff
@@ -705,19 +705,15 @@ private Map<String, Object> getMessages(IMessageBrowser<?> transactionalStorage,
 						continue;
 
 					count++;
-					if (count > filter.skipMessages()) { 
+					if (count > filter.skipMessages() && messages.size() < filter.maxMessages()) {
 						StorageItemDTO dto = new StorageItemDTO(iterItem);
 						dto.setPosition(count);
 						messages.add(dto);
 					}
```

Comment:
now it doesnt break anymore, so it will continue to iterate over all items in the storage :(

Reply:
Yes, unfortunately. It is required to find how many messages exist after filtering to display the numbers correctly at the bottom of the table.
Below image shows it correctly after this change.
![image](https://user-images.githubusercontent.com/10929856/163380916-c4a5949a-caa4-473a-95c5-18c9a9e994d9.png)

Here is a faulty one. After applying filter there are 74 messages with maxMessages 100. 
![image](https://user-images.githubusercontent.com/10929856/163381394-1667a461-1fc2-41ac-b2f7-7e9617e1f0de.png)